ee/desktop/runtime/runner.go

```diff
@@ -31,12 +33,41 @@ type DesktopUsersProcessesRunner struct {
 	executablePath string
 }
 
+func (r *DesktopUsersProcessesRunner) addProcessForUser(uid string, osProcess *os.Process) error {
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second*2)
+	defer cancel()
+
+	psutilProc, err := process.NewProcessWithContext(ctx, int32(osProcess.Pid))
+	if err != nil {
+		return fmt.Errorf("creating process record: %w", err)
+	}
+
+	path, err := psutilProc.ExeWithContext(ctx)
+	if err != nil {
+		return fmt.Errorf("getting process path: %w", err)
+	}
+
+	r.uidProcs[uid] = processRecord{
```

Comment:
Is this racy? I feel like modifying hashes always need a mutex.

Reply:
this is actually not accessed concurrently, we iterate over users sequentially and add them here